developing-packages.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: page
+title: Developing Packages & Plugins
+permalink: /developing-packages/
+---
+
+* TOC
+{:toc}
+
+## Package introduction
+
+Dart packages enable the creation of modular code that can easily be shared. A
+minimal package with a single library consists of:
+
+* A `pubspec.yaml` file: A metadata file that declared the package name,
+  version, author, etc.
+
+* A `lib` directory containing the public code in the package, minimally a
+  single `<package-name>.dart` file.
+
+For additional details, see the [Dart library package](https://www.dartlang.org/guides/libraries/create-library-packages) documentation.
+
+### Package types {#types}
+
+Flutter supports several kinds of packages:
```

Comment:
I think we're being confusing here. There is one type of package: a pub package (or, call it a Dart package). It can contain many different kinds of things: Dart code, plugin code, or a combination.

I agree that package developers might want to know the different kinds of things they can put in it.

If you're willing to change this phrasing to something like:

* Flutter uses pub for its package format and distribution.
* A pub (Dart) package can contain many things: Dart code, plugin code, example code, assets, and more.

Maybe it's as simple as "Dart packages can contain several kinds of code: Dart, plugin"

So, it's not about supporting different kinds of packages, it's that there are different kinds of contents.